utils/src/base58.rs

```diff
@@ -0,0 +1,118 @@
+// from: https://github.com/Nullus157/bs58-rs/blob/main/src/encode.rs
+pub const ALPHABET: &'static [u8] = b"123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+
+pub fn base58_encode_inputs<'a>(inputs: &[&[u8]], output: &'a mut [u8]) -> Option<&'a [u8]> {
```

Comment:
It's used here: https://github.com/alephium/ledger-alephium/blob/efb97560a2e18184a85490245d1f46a1cb47b150/app/src/tx_reviewer.rs#L589

Because we will only keep the hash after decoding, we need to add the prefix when encoding `P2PKH` and `P2SH` lookup script. If we want to use `&[u8]`, we have to allocate a 33-byte array, copy the prefix and hash into it, and then encode.